File: AndroidDemoProject/app/src/main/java/com/by/androiddemoproject/bridges/flutter/FlutterEngineManager.kt
package com.by.androiddemoproject.bridges.flutter

import android.content.Context

object FlutterEngineManager {
    private var isPreWarmed = false

    // Call in Application.onCreate()
    // Real implementation:
    // val engine = FlutterEngine(context)
    // engine.dartExecutor.executeDartEntrypoint(DartExecutor.DartEntrypoint.createDefault())
    // FlutterEngineCache.getInstance().put("main_engine", engine)
    fun preWarm(context: Context) {
        isPreWarmed = true
        println("[FlutterEngineManager] Engine pre-warmed")
    }

    val engineReady: Boolean get() = isPreWarmed
}

File: iOSDemoProject/iOSDemoProject/Bridges/FlutterBridge/FlutterEngineManager.swift
import UIKit

final class FlutterEngineManager {
    static let shared = FlutterEngineManager()

    private var isPreWarmed = false

    private init() {}

    /// Call in AppDelegate.didFinishLaunchingWithOptions to pre-warm the Flutter engine
    /// In real implementation, this would create and run a FlutterEngine instance
    func preWarm() {
        // Real implementation:
        // let engine = FlutterEngine(name: "main_engine")
        // engine.run()
        // FlutterEngineCache.shared.cache(engine, key: "main_engine")
        isPreWarmed = true
        print("[FlutterEngineManager] Engine pre-warmed")
    }

    var engineReady: Bool { isPreWarmed }
}
